### DISCUSSION TITLE
Trivy Dockerhub and Personal Access Tokens

### DISCUSSION THREAD

**Discussion by chaoscode** [aquasecurity/trivy#7322] [Q&A] [answered]
### Question

I've had some issues getting the cli to work with a PAT (Personal Access Tokens) for private docker images hosted on dockerhub. I've tried a few combos with the username and password flags and also the token flag. Does anyone have a working example of the cli call to scan a private image on dockerhub with a PAT?

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Operating System

Linux - ParrotOS v15

### Version

```bash
Version: 0.54.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-08-08 18:16:30.613959512 +0000 UTC
  NextUpdate: 2024-08-09 00:16:30.613959191 +0000 UTC
  DownloadedAt: 2024-08-08 21:33:06.050285802 +0000 UTC
```

**Comment by knqyf263** [comment, marked as answer]
Did you try `docker login`? If you successfully get logged in, Trivy should work.

**Reply by chaoscode** [reply]
I will give that a shot

**Reply by chaoscode** [reply]
it was a pain to get docker installed inside a docker but I got it working. docker login does work with PAT (Personal Access Token)

**Reply by knqyf263** [reply]
If you pass the same credential, where `docker login` works, to Trivy, it should work.
https://aquasecurity.github.io/trivy/v0.54/docs/advanced/private-registries/
